### PR TITLE
WIP - cats-effect, fs2 0.10

### PR DESCRIPTION
**Pull request by tpolecat**
This sets the stage for the 0.5.x series. See `CHANGELOG.md` for an overview of changes.

A few remaining things.

- [x] fix `Transactor.exec` and `Coproduct.scala` and `StreamingCopy.scala`
- [x] why is the build not building tut doc?
- [x] ~~monix example in `03-Connecting.md`~~ deferred as #567 

I will release 0.5.0-M1 once this is merged.

*This is a fantastically gigantic refactor, and it's a great testament to the power of FP and static types that other than one dumb error there have been no test failures during this process. As soon as stuff compiled it worked.*

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -64,7 +73,7 @@ This is probably the most important development for 0.4.0 and it is due in large
 - The **dynamic SQL** story is now slightly better with the introduction of **composable statement fragments**. These allow you to build statements from smaller pieces without having to track parameter placeholders/offsets by hand. See the book chapter for details and examples.
 - SQL `IN` clauses are now handled via `Fragments.in`, which is a **breaking change** relative to 0.3.0. See the book chapter on parameterized queries for an example.
 - Methods on `Query[0]/Update[0]` that construct streams (`.process`) now have variants that allow you to specify the **chunk size**, which by default is 512 rows.
-- There is now an `IOLite` data type that you can use if you're having a hard time settling on a target effect type. It works identically in Cats and scalaz and is what's used in the book.
+- There is now an `IO` data type that you can use if you're having a hard time settling on a target effect type. It works identically in Cats and scalaz and is what's used in the book.
```
Comment (gatorcse) [inline CHANGELOG.md]: Mentions scalaz, should this be removed?

Comment (tpolecat) [inline CHANGELOG.md]: CHANGELOG will include historical information that's been superseded.

**Comment by tpolecat** [comment]
@wedens I had to remove your `fuseMap2` implementation from `StreamingCopy.scala` because you're doing `Stream.translate` into a kleisli arrow which you can't do anymore … the target types needs `Effect` now which means you have to be able to go `F[A] => IO[A]`. Just FYI … it's example code so it's not a big deal, but if you can think of a fix I'm happy to put it back.

**Comment by wedens** [comment]
@tpolecat It's unfortunate consequence of https://github.com/typelevel/cats/issues/1733 :(

**Pull request merged**